app/src/main/java/com/rahobbs/todo/fragments/TodoFragment.java

```diff
@@ -0,0 +1,228 @@
+import android.view.MenuItem;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.CheckBox;
+import android.widget.CompoundButton;
+import android.widget.EditText;
+import android.widget.LinearLayout;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import com.firebase.client.Firebase;
+import com.rahobbs.todo.helpers.Feedback;
+import com.rahobbs.todo.R;
+import com.rahobbs.todo.helpers.TodoItem;
+import com.rahobbs.todo.helpers.TodoLab;
+
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Locale;
+import java.util.UUID;
+
+/**
+ * Created by rachael on 2/1/16.
+ * Fragment that contains a single to-do item detailed view
+ */
+public class TodoFragment extends Fragment {
+
+    private static final String ARG_TODO_ID = "todoId";
+    private static final String DIALOG_DATE = "DialogDate";
+    private static final int REQUEST_DATE = 0;
+
+    private TodoItem mTodo;
+    private EditText mTitle;
+    private EditText mDetails;
+    private TextView mDueDateTextField;
+    private CheckBox mCompletedCheckbox;
+    private LinearLayout mDateComponents;
+
+    public static TodoFragment newInstance(UUID todoId) {
+        Bundle args = new Bundle();
+        args.putSerializable(ARG_TODO_ID, todoId);
+        TodoFragment fragment = new TodoFragment();
+        fragment.setArguments(args);
+        return fragment;
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        Firebase.setAndroidContext(getContext());
+        UUID todoId = (UUID) getArguments().getSerializable(ARG_TODO_ID);
+        setHasOptionsMenu(true);
+
+        if (todoId == null) {
+            throw new RuntimeException("Todo ID is null.");
+        }
+
+        mTodo = TodoLab.get(getActivity()).getItem(todoId);
+        Log.v("DETAIL_CREATED", mTodo.getID().toString());
+
+        if (mTodo == null) {
+            throw new RuntimeException("No Todo item with provided ID (" + todoId.toString() + ") was found.");
+        }
+    }
+
+    @Override
+    public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
+        super.onCreateOptionsMenu(menu, inflater);
+        inflater.inflate(R.menu.fragment_todo_item, menu);
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        switch (item.getItemId()) {
+            case R.id.menu_item_send_feedback:
+                Intent i = new Intent(Intent.ACTION_SEND);
+                Feedback fb = new Feedback();
+                fb.sendFeedback(i);
+                startActivity(Intent.createChooser(i, "Send mail..."));
+                return true;
+            case R.id.menu_item_delete_todo:
+                UUID todoId = mTodo.getID();
+                TodoLab.get(getActivity()).deleteTodoItem(todoId);
+
+                String titleToDisplay;
+                if (mTodo.getTitle() == null) {
```

Comment:
it looks like were getting into some messy state issues here...

The Fragment has a variable for the `TodoItem`. It is alive as long as the fragment is alive. However, a few lines up you've deleted the `TodoItem` from the database. But it's still exists as `mTodo` in this fragment. Hello, weird state!

You want to avoid things like this and make sure you are always thinking about "what is the truth". In this case, what is the truth about this `TodoItem` does it exist, or not? and the answer should be the same everywhere in the app.